planex/spec.py

```diff
@@ -62,19 +64,25 @@ def __init__(self, path, dist="", check_package_name=True, topdir=None):
         # match the files actually produced by mock.
         rpm.addMacro('dist', dist)
 
-        try:
-            # silence errors about missing sources
-            with open(os.devnull, "w") as nullfh:
+        with tempfile.TemporaryFile() as nullfh:
+            try:
+                # collect all output to stderr then filter out
+                # errors about missing sources
                 errcpy = os.dup(2)
                 try:
                     os.dup2(nullfh.fileno(), 2)
                     self.spec = rpm.ts().parseSpec(path)
                 finally:
                     os.dup2(errcpy, 2)
                     os.close(errcpy)
-        except ValueError as exn:
-            exn.args = (exn.args[0].rstrip() + ' ' + path, )
-            raise
+            except ValueError as exn:
+                nullfh.seek(0, os.SEEK_SET)
+                for line in nullfh:
+                    line = line.strip()
+                    if not line.endswith(': No such file or directory'):
+                        print >>sys.stderr, line
+                exn.args = (exn.args[0].rstrip() + ' ' + path, )
```

Comment:
I'm not a Python expert, but `exn.args = (<expression>, )` looks strange with just nothing in the second component.